packages/storage/src/providers/AWSS3Provider.ts

```diff
@@ -539,29 +565,31 @@ export class AWSS3Provider implements StorageProvider {
 		}
 	}
 
-	/**
-	 * @private
-	 */
-	_ensureCredentials() {
-		return Credentials.get()
-			.then(credentials => {
-				if (!credentials) return false;
-				const cred = Credentials.shear(credentials);
-				logger.debug('set credentials for storage', cred);
-				this._config.credentials = cred;
-
-				return true;
-			})
-			.catch(error => {
-				logger.warn('ensure credentials error', error);
-				return false;
-			});
+	private async _ensureCredentials(): Promise<boolean> {
+		try {
+			const credentials = await Credentials.get();
+			if (!credentials) return false;
+			const cred = Credentials.shear(credentials);
+			logger.debug('set credentials for storage', cred);
+			this._config.credentials = cred;
+
+			return true;
+		} catch (error) {
+			logger.warn('ensure credentials error', error);
+			return false;
+		}
 	}
 
-	/**
-	 * @private
-	 */
-	private _prefix(config): string {
+	private _isWithCredentials(config: StorageOptions): config is StorageOptions & { credentials: ICredentials } {
+		return typeof config === 'object' && config.hasOwnProperty('credentials');
+	}
+
+	private _prefix(config: {
+		credentials: ICredentials;
+		level?: StorageLevel;
+		customPrefix?: CustomPrefix;
+		identityId?: string;
+	}) {
```

Comment:
It is better to be explicit in the return types instead of relying on TypeScript's inference as changes in implementation could inadvertently change the return type of the function.

```suggestion
	}): string {
```

Reply:
There are other places in the PR where we could do this too